mapstore.go

```diff
@@ -20,38 +22,64 @@ func (e *InvalidKeyError) Error() string {
 	return fmt.Sprintf("invalid key: %x", e.Key)
 }
 
+// ErrWrongKeySize is returned when a key has a different size than the key size.
+var ErrWrongKeySize = errors.New("wrong key size")
+
 // SimpleMap is a simple in-memory map.
 type SimpleMap struct {
-	m map[string][]byte
+	m       map[string][]byte
+	keySize int
 }
 
 // NewSimpleMap creates a new empty SimpleMap.
-func NewSimpleMap() *SimpleMap {
+func NewSimpleMap(keySize int) *SimpleMap {
 	return &SimpleMap{
-		m: make(map[string][]byte),
+		m:       make(map[string][]byte),
+		keySize: keySize,
```

Comment:
Should this throw an error if the keysize is less than `1`?

Reply:
absolutely. Will make the change. Thanks